vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -1,126 +1,282 @@
+        return f"{self.__class__.__name__}({self._name!r}, {self.__load_data.__qualname__})"
+
+
+class _StaticData:
+    """Wrapper for a pd.DataFrame. This data cannot be updated during runtime.
+
+    This is currently private since it's not expected that a user would instantiate it directly. Instead, you would use
+    through the data_manager interface as follows:
+         >>> data_manager["static_data"] = pd.read_csv("static_data.csv")
+
+    This class does not have much functionality but exists for a couple of reasons:
+        1. to align interface with _DynamicData by providing a load method so that fetching data from data_manager can
+        transparently handle loading both _StaticData and _DynamicData without needing switching logic
+        2. to raise a clear error message if a user tries to set a timeout on the data source
+    """
+
+    def __init__(self, data: pd.DataFrame):
+        # No need for _name here because static data doesn't need a cache key.
+        self.__data = data
+
+    def load(self) -> pd.DataFrame:
+        """Loads data.
+
+        Returns a copy of the data. This  is not necessary if we are careful to not do any inplace=True operations,
+        but probably safest to leave it here.
+        """
+        return self.__data.copy()
+
+    def __setattr__(self, name, value):
+        # Any attributes that are only relevant for _DynamicData should go here to raise a clear error message.
+        if name in ["timeout"]:
+            raise AttributeError(
+                f"Static data that is a pandas.DataFrame itself does not support {name}; you should instead use a "
+                "dynamic data source that is a function that returns a pandas.DataFrame."
+            )
+        super().__setattr__(name, value)
 
 
 class DataManager:
     """Object to handle all data for the `vizro` application.
 
     Examples
-        >>> import plotly.express as px
-        >>> data_manager["iris"] = px.data.iris()
+        >>> # Static data that cannot be refreshed during runtime
+        >>> data_manager["data"] = pd.read_csv("data.csv")
+        >>> # Data that can be refreshed during runtime
+        >>> def dynamic_data():
+        >>>     return pd.read_csv("dynamic_data.csv")
+        >>> data_manager["dynamic_data"] = dynamic_data
+        >>> data_manager["dynamic_data"].timeout = 5  # if you want to change the cache timeout to 5 seconds
 
     """
 
     def __init__(self):
-        self.__lazy_data: Dict[DatasetName, pd_LazyDataFrame] = {}
-        self.__original_data: Dict[DatasetName, pd.DataFrame] = {}
-        self.__component_to_original: Dict[ComponentID, DatasetName] = {}
+        self.__data: Dict[DataSourceName, Union[_DynamicData, _StaticData]] = {}
+        self.__component_to_data: Dict[ComponentID, DataSourceName] = {}
         self._frozen_state = False
+        self.cache = Cache(config={"CACHE_TYPE": "NullCache"})
```

Comment:
@l0uden noticed the same thing and I added a note to our docs explaining this - see the last paragraph in https://vizro--398.org.readthedocs.build/en/398/pages/user-guides/dynamic-data/#configure-cache.

I spent a looong time yesterday thinking about whether we should clear the cache when `Vizro` or `data_manager` is instantiated and decided this is a bad idea. Flask-caching is designed to not be cleared up when the process is killed. Doing so would actually have some disadvantages e.g. it makes it hard to restart an individual worker in a production environment and it means users don't benefit from speedup while developing. It's better to leave it uncleared by default and having some orphaned files or RAM feels a bit untidy but is ok.

Reply:
Ah cool, I didn't notice it was added to the docs. 🚀 

I agree that we don't clear the cache when Vizro or data_manager is instantiated, as long as it's described in the docs 😄.
The only thing that might be worth considering is adding and exposing the `clear_cache: bool = False` option somewhere. But, it could/should also be part of some other PR if we determine that it is a useful feature.